tests/agent/test_k8s_agent.py

```diff
@@ -213,3 +216,70 @@ def test_k8s_agent_generate_deployment_yaml_no_resource_manager(
     assert agent_env[2]["value"] == "test_namespace"
 
     assert len(deployment["spec"]["template"]["spec"]["containers"]) == 1
+
+
+def test_k8s_agent_heartbeat_creates_file(monkeypatch, runner_token):
+    k8s_config = MagicMock()
+    monkeypatch.setattr("kubernetes.config", k8s_config)
+
+    with tempfile.TemporaryDirectory() as tempdir:
+
+        monkeypatch.setattr(
+            "prefect.agent.kubernetes.agent.AGENT_DIRECTORY",
+            "{}/.prefect/agent".format(tempdir),
```

Comment:
So these tests will fail on Windows :/ but I can worry about updating them in a follow-up PR

Reply:
Thanks! Haha